src/Compilers/CSharp/Portable/Parser/Lexer_RawStringLiteral.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
+{
+    internal partial class Lexer
+    {
+        /// <returns>The number of quotes that were consumed</returns>
+        private int ConsumeQuoteSequence()
+        {
+            var start = TextWindow.Position;
+            while (TextWindow.PeekChar() == '"')
+                TextWindow.AdvanceChar();
+
+            return TextWindow.Position - start;
+        }
+
+        private void ConsumeWhitespace(StringBuilder? builder)
+        {
+            while (true)
+            {
+                var ch = TextWindow.PeekChar();
+                if (!SyntaxFacts.IsWhitespace(ch))
+                    break;
+
+                builder?.Append(ch);
+                TextWindow.AdvanceChar();
+            }
+        }
+
+        private bool IsAtEndOfText(char currentChar)
+            => currentChar == SlidingTextWindow.InvalidCharacter && TextWindow.IsReallyAtEnd();
+
+        private void ScanRawStringLiteral(ref TokenInfo info)
+        {
+            _builder.Length = 0;
+
+            var startingQuoteCount = ConsumeQuoteSequence();
+
+            Debug.Assert(startingQuoteCount >= 3);
+
+            // Keep consuming whitespace after the initial quote sequence.
+            ConsumeWhitespace(builder: null);
+
+            if (SyntaxFacts.IsNewLine(TextWindow.PeekChar()))
+            {
+                // Past the initial whitespace, and we hit a newline, this is a multi line raw string literal.
+                ScanMultiLineRawStringLiteral(ref info, startingQuoteCount);
+            }
+            else
+            {
+                // Past the initial whitespace, and we hit anything else, this is a single line raw string literal.
+                ScanSingleLineRawStringLiteral(ref info, startingQuoteCount);
+            }
+
+            // If we encounter any errors while scanning this raw string, then always treat its constant value
+            // as unknown.
+            if (this.HasErrors)
```

Comment:
I didn't see an equivalent to this check in Lexer_StringLiteral.cs. Do we set a similar constant value for literal strings with errors?

Reply:
i believe so, but i will verify and circle back.

Reply:
Nope.  you're correct.  normal strings still contain content even with errors.  I'll update to have the same behavior.